release-notes.txt

```diff
@@ -1,3 +1,270 @@
+    + atmega_common: allow defining rom and ram length for link
+    * cortexm_common/ldscript: simplify generation of firmwares in section of
+      the ROM
+    * cpu/stm32_common: fix source selection declared as module dependencies
+
+Device Drivers
+--------------
+    + drivers/vcnl40x10: added initial support
+    * drivers/periph_i2c: rework API
+    * drivers/*: unify driver initialization parameters scheme
+    * drivers/ata8520e: rework internal implementation
+
+Build System / Tooling
+----------------------
+    * pyterm: handle ctrl+d shortcut nicely
+    + make: process include and dep for external modules
+    + docker: allow using sudo for docker
+    + docker: Use system GIT_CACHE_DIR if available
+    * docker: export RIOT_CI_BUILD_to docker
+    * docker: handle building in git worktree
+    * docker/Makefile.include: Fix BUILDRELPATH when RIOTPROJECT is CURDIR
+    * pkg.mk: do not use user identidy when applying patches
+    * makefiles/buildtests.inc.mk: fix 'clean-intermediates' not cleaning
+    * llvm.inc.mk: completely replace detection of GCC includes
+
+Testing
+-------
+    + automatic compile tests for LLVM for selected boards.
+    + tests/periph_uart: add automated testing script
+    + tests/periph_eeprom: add automated testing script
+    + tests/periph_flashpage: add automated testing script
+
+Known Issues
+============
+
+Networking related issues
+-------------------------
+
+#4527: gnrc_ipv6: Multicast is not forwarded if routing node listens to the
+       address
+#5016: gnrc_rpl: Rejoining RPL instance as root after reboot messes up rooting
+#5051: Forwarding a packet back to its link layer source should not be allowed
+#5230: gnrc_ipv6: multicast packes are not dispatched to the upper layers
+#5748: gnrc: nodes crashing with too small packet buffer
+#6123: gnrc: crash with (excessive) traffic in native
+#7474: 6lo: gnrc fragmentation expects driver to block on TX
+#7737: pkg: libcoap is partially broken and outdated
+#8086: gnrc_rpl_p2p: not compiling
+#8130: gcoap: can't build witth network stacks other than gnrc
+#8172: gnrc_netif, gnrc_uhcpc: Replacing prefix on border router results in no
+       configured prefix
+#8271: app/netdev: application stops working after receiving frames with
+       assertion or completely without error
+#9616: at86rf2xx: can not correctly set channel after `NETOPT_STATE_RESET`
+#9656: gnrc/netif: various problems after resetting interface a second time
+#9584: drivers/cc2420: add netopt NETOPT_MAX_PACKET_SIZE
+
+Timer related issues
+--------------------
+
+#5103: xtimer: weird behavior of tests/xtimer_drift
+#5338: xtimer: xtimer_now() is not ISR safe
+#6052: tests: xtimer_drift_gets_stuck
+#7114: xtimer: adds items to the wrong list if the timer overflows between
+       _xtimer_now() and irq_disable()
+#7347: xtimer_usleep stuck for small values
+#8251: telosb: timer config wrong when running o a tmote sky
+#9052: misc issues with tests/trickle
```

Comment:
#10073: timer_usleep wrong for small delay times (https://github.com/RIOT-OS/RIOT/issues/10073)